src/config/variables.go

```diff
@@ -0,0 +1,119 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/defenseunicorns/zarf/src/internal/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/pterm/pterm"
+)
+
+// FillActiveTemplate handles setting the active variables and reloading the base template.
+func FillActiveTemplate() error {
+	packageVariables, err := utils.FindYamlTemplates(&active, "###ZARF_PKG_VAR_", "###")
+	if err != nil {
+		return err
+	}
+
+	for key := range CommonOptions.SetVariables {
+		value := CommonOptions.SetVariables[key]
+		packageVariables[key] = &value
+	}
+
+	for key, value := range packageVariables {
+		if value == nil && !CommonOptions.Confirm {
+			setVal, err := promptVariable(key, nil)
+
+			if err == nil {
+				packageVariables[key] = &setVal
+			} else {
+				return err
+			}
+		} else if value == nil {
+			return fmt.Errorf("variable '%s' must be '--set' when using the '--confirm' flag", key)
+		}
+	}
+
+	templateMap := map[string]string{}
+	for key, value := range packageVariables {
+		// Variable keys are always uppercase in the format ###ZARF_PKG_VAR_KEY###
+		templateMap[strings.ToUpper(fmt.Sprintf("###ZARF_PKG_VAR_%s###", key))] = *value
+	}
+
+	return utils.ReloadYamlTemplate(&active, templateMap)
+}
+
+// SetActiveVariables handles setting the active variables used to template component files.
+func SetActiveVariables() error {
+	SetVariableMap = CommonOptions.SetVariables
+
+	for _, variable := range active.Variables {
```

Comment:
This reads a little odd to me with all the multiple negative conditins.  Some rewrite suggestions:

```golang
func SetActiveVariables() error {
	SetVariableMap = CommonOptions.SetVariables

	for _, variable := range active.Variables {
		_, present := SetVariableMap[variable.Name]

		// Variable is present, no need to continue checking
		if present {
			continue
		}

		// First set default if it exists
		if variable.Default != nil {
			SetVariableMap[variable.Name] = *variable.Default
		}

		// Prompt is enabled
		if variable.Prompt {
			// If Prompt and Confirm are set, throw an error
			if CommonOptions.Confirm {
				return fmt.Errorf("variable '%s' must be '--set' when using the '--confirm' flag", variable.Name)	
			}

			// Prompt the user for input 
			SetVariableMap[variable.Name], err := promptVariable(variable.Name, variable.Default)
			if err != nil {
				return err
			} 
		}
	}

	return nil
}
```

Reply:
Refactored to something close to this.  Definitely agree it is better.

Reply:
Removed the confirm check since default will always be set now.